### PR TITLE
KOGITO-5715: Online DMN Editor should support deployment to any Openshift Cluster other than Dev Sandbox

### PR DESCRIPTION
**Pull request by rafaeltuelho**
This PR introduces a change to adapt the DMN deployment capability to support any Openshift Cluster instead of only supporting Red Hat Dev Sandbox.

Merge with: https://github.com/kiegroup/kogito-tooling-go/pull/21

### CODE REVIEW

**Review by porcelli — CHANGES_REQUESTED** [review]
thx @rafaeltuelho for the contribution! this is great. 

before a real round of review, could you please clean the noise in the PR, like: 
- the versioning assignment
- re-ordering of config files

**Comment by rafaeltuelho** [comment]
> thx @rafaeltuelho for the contribution! this is great.
> 
> before a real round of review, could you please clean the noise in the PR, like:
> 
> * the versioning assignment
> * re-ordering of config files

Sure I'll take care of this. thanks

**Comment by porcelli** [comment]
thx very much @rafaeltuelho for the update!

**Comment by rafaeltuelho** [comment]
> thx very much @rafaeltuelho for the update!

Sure thing! Thanks @tiagobento and @caponetto for their patience teaching me some git magic tricks :-)

File: packages/online-editor/src/common/i18n/locales/en.ts [hunk 1]
```diff
@@ -164,7 +165,7 @@ export const en: OnlineI18n = {
           connectionSuccess: "Connection successfully established.",
           connectionError: "Connection refused.",
           introduction: "Now you are able to:",
-          deployNowExplanation: `${"Deplow now".bold()}: Your model will be deployed on your instance. This action will take a few minutes to be completed and you will need to create a new deployment if you update your model.`,
+          deployNowExplanation: `${"Deploy now".bold()}: Your model will be deployed on your instance. This action will take a few minutes to be completed and you will need to create a new deployment if you update your model.`,
```
Comment (caponetto) [inline packages/online-editor/src/common/i18n/locales/en.ts:168]: Thanks for fixing it!

File: packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ export const DEVELOPER_SANDBOX_URL = "https://developers.redhat.com/developer-sa
 export const DEVELOPER_SANDBOX_GET_STARTED_URL = "https://developers.redhat.com/developer-sandbox/get-started";
 
 export class DmnDevSandboxService {
-  private readonly RESOURCE_NAME_PREFIX = "dmn-dev-sandbox";
+  private readonly RESOURCE_NAME_PREFIX = "dmn-service";
```
Comment (caponetto) [inline packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx]: IMO we can keep the `dmn-dev-sandbox` prefix as this is the name of the feature, i.e. not directly related to Openshift Sandbox. cc @tiagobento

File: packages/i18n-common-dictionary/src/CommonI18n.ts [hunk 3]
```diff
@@ -73,6 +73,7 @@ export type CommonDictionary = {
   uninstall: string;
   username: string;
   validation: string;
+  namespace: string;
```
Comment (caponetto) [inline packages/i18n-common-dictionary/src/CommonI18n.ts]: This list of properties is in alphabetical order. Can you place `namespace` before `new`?

File: packages/i18n-common-dictionary/src/locales/en.ts [hunk 4]
```diff
@@ -75,5 +75,6 @@ export const en: CommonI18n = {
     uninstall: "Uninstall",
     username: "Username",
     validation: "Validation",
+    namespace: "Namespace (project)",
```
Comment (caponetto) [inline packages/i18n-common-dictionary/src/locales/en.ts]: This list of properties is in alphabetical order. Can you place `namespace` before `new`?

File: packages/online-editor/src/common/i18n/locales/en.ts [hunk 5]
```diff
@@ -145,9 +145,10 @@ export const en: OnlineI18n = {
           introduction: `In order to create your ${en_common.names.shortDevSandbox} instance:`,
           goToGetStartedPage: "Go to the Get Started page",
           followSteps: `Follow the steps to launch your instance. You will be asked to log in with your ${en_common.names.redHat} account.`,
-          informUsername: `Once your instance is up and running, inform the username of your ${en_common.names.redHat} account below.`,
-          inputReason: "This information is necessary for locating your namespaces.",
-          usernamePlaceholder: `The username of your ${en_common.names.redHat} account`,
+          informNamespace: `Once your instance is up and running, inform the namespace (project) in your cluster you want to have the Decision Model deployed to.`,
+          inputReason:
+            "This information is necessary for deploying youde Decision Model into the right project namespace.",
```
Comment (caponetto) [inline packages/online-editor/src/common/i18n/locales/en.ts]:
```suggestion
            "This information is necessary for deploying your Decision Model into the right project namespace.",
```
There is a typo here.

File: packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ export const DEVELOPER_SANDBOX_URL = "https://developers.redhat.com/developer-sa
...
   private readonly NAMESPACE_SUFFIX = "-dev";
```
Comment (caponetto) [inline packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx]: Could you please remove the `NAMESPACE_SUFFIX` constant and the  `composeNamespace` method?
With your changes, they no longer need to be there.

**Review by caponetto — COMMENTED** [review]
Hi @rafaeltuelho, thanks for the initiative, I really appreciate it!
I've left some minor comments, please check if they make sense.

File: packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx (hunk 2 above)
Comment (rafaeltuelho) [reply packages/online-editor/src/editor/DmnDevSandbox/DmnDevSandboxService.tsx]: Sure, that makes sense.

File: packages/i18n-common-dictionary/src/CommonI18n.ts (hunk 3 above)
Comment (rafaeltuelho) [reply packages/i18n-common-dictionary/src/CommonI18n.ts]: Got it. Just out of curiosity, does this influence the runtime or is it just a code convention?

File: packages/online-editor/src/common/i18n/locales/en.ts (hunk 5 above)
Comment (rafaeltuelho) [reply packages/online-editor/src/common/i18n/locales/en.ts]: thanks

File: packages/i18n-common-dictionary/src/CommonI18n.ts (hunk 3 above)
Comment (caponetto) [reply packages/i18n-common-dictionary/src/CommonI18n.ts]: It is just a convention.

**Comment by caponetto** [comment]
Hi @tomasdavidorg, can you please take a look at this PR when possible?

We only use the `username` to append with `-dev` and deploy to `username-dev` namespace.
In this PR, @rafaeltuelho changed the required config from `username` to `namespace` to make it more flexible.

Anyway, the testing goal is just to make sure that the Openshift Sandbox, which is the one that we currently entirely support, is still ok after these changes. I have tested and everything is good to go. 

Thanks!

**Review by caponetto — APPROVED** [review]
Thanks, @rafaeltuelho! :)

**Comment by tomasdavidorg** [comment]
Hi @caponetto , I was not able to setup the connection with the OpenShift sandbox. Do I need some specific configuration / build parameters?
I built the branch and started the server by:
```
export ONLINE_EDITOR__kieToolingExtendedServicesCompatibleVersion=0.12.1
yarn bootstrap && yarn build:dev
cd packages/online-editor
yarn start
```

After I put the login informations to the modal dialog, I got `Connection refused. Please check the information provided.`.

Do I have to setup `export DMN_DEV_SANDBOX__baseImageAccount` or any other env. variable?

**Review by tomasdavidorg — APPROVED** [review]
Thanks @caponetto . The issue was that I open the application from http://localhost:9001/ and not from http://0.0.0.0:9001/. Approving.

**Comment by caponetto** [comment]
@porcelli can we go ahead and merge this PR?

**Review by porcelli — APPROVED** [review]

**Review by ederign — APPROVED** [review]

**Pull request merged**